pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -615,5 +604,15 @@ pub mod pallet {
 				} => Self::process_outbound_message(sender, destination, message),
 			}
 		}
+
+		/// Process a message.
+		fn max_processing_weight(msg: &Self::Message) -> Weight {
+			match msg {
+				GatewayMessage::Inbound { message, .. } => {
+					LP_DEFENSIVE_WEIGHT.saturating_mul(message.submessages().len() as u64)
```

Comment:
Nit: For production code, I would like to avoid casting with `as` and rather use the defensive `saturated_into`.

Reply:
Only when converting from `usize` to `u64`, I use `as` because our machine target is no more than `u64` (are they machines longer than that?) 

But it seems reasonable; I'll add it to my mental list of changes!